UI/MapWnd.cpp

```diff
@@ -2781,16 +2781,91 @@ void MapWnd::ClearSystemRenderingBuffers() {
     m_star_circle_vertices.clear();
 }
 
-namespace {
+namespace GetPathsThroughSupplyLanes {
     // SupplyLaneMap map keyed by system containing all systems
     // corresponding to valid supply lane destinations
     typedef boost::unordered_multimap<int,int> SupplyLaneMMap;
 
-    // AncestorInfo stores the \p ids of systems one back on the path.
-    struct AncestorInfo {
-        AncestorInfo(int a, int o) : ids(1, a), single_origin(o) {}
-        std::vector<int> ids;  // previous systems
-        // originating terminal or nothing for connected resources
+
+    /**
+       GetPathsThroughSupplyLanes starts with:
+
+       \p terminal_points are system ids of systems that contain either
+       a resource source or a resource sink.
+
+       \p supply_lanes are pairs of system ids at the end of supply
+       lanes.
+
+       GetPathsThroughSupplyLanes returns a \p good_path.
+
+       The \p good_path is all system ids of systems connecting any \p
+       terminal_point to any other reachable \p terminal_point along the
+       \p supply_lanes. The \p good_path is all systems that could
+       transport a resource from a source to a sink along a starlane
+       that is part of the supply. The \p good_path includes the
+       terminal point system ids that are part of the network.  The \p
```

Comment:
What is "the network"? Different from other terms for groups of connected systems used above?